### PR TITLE
{Search} Add RBAC command support to Search

### PR DESCRIPTION
**Pull request by mattmsft**
**Related command**

az search

**Description**

Adding support for RBAC to the search command
https://learn.microsoft.com/en-us/azure/search/search-security-rbac?tabs=config-svc-portal%2Croles-portal%2Ctest-portal%2Ccustom-role-portal%2Cdisable-keys-portal

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by mattmsft** [comment]
This is broken right now - can you help me fix this?

**Comment by yonzhan** [comment]
@zhoxing-ms for awareness

File: src/azure-cli/azure/cli/command_modules/search/custom.py [hunk 1]
```diff
@@ -36,8 +37,16 @@ def create_search_service(cmd, resource_group_name, search_service_name, sku, lo
...
+    :param disable_local_auth: If calls to the search service will not be permitted to utilize API keys for authentication;
+                     This cannot be combined with auth_options.
+    :param auth_options: Options for authenticating calls to the search service; possible values include "aadOrApiKey", "apiKeyOnly";
+                     This cannot be combined with disable_local_auth.
+    :param aad_auth_failure_mode: Describes response code from calls to the search service that failed authentication; possible values include "http401WithBearerChallenge", "http403";
+                     This cannot be combined with disable_local_auth.
     """
-    from azure.mgmt.search.models import SearchService, Sku, NetworkRuleSet, IpRule, Identity
+    SearchService, Sku, NetworkRuleSet, IpRule, Identity = cmd.get_models('SearchService', 'Sku', 'NetworkRuleSet', 'IpRule', 'Identity')
+    if SearchService is None:
```
Comment (mattmsft) [inline src/azure-cli/azure/cli/command_modules/search/custom.py:48]: why is this None?

**Comment by jsntcy** [comment]
@yanzhudd, could you please help to take a look the issue above?

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**